Pod/Classes/VOKCoreDataManager.m

```diff
@@ -319,46 +296,20 @@ - (NSArray *)importArray:(NSArray *)inputArray forClass:(Class)objectClass withC
         } else {
             returnObject = [self managedObjectOfClass:objectClass inContext:contextOrNil];
             [mapper setInformationFromDictionary:inputDict forManagedObject:returnObject];
+            if (inputKey) {
+                [existingObjectArray addObject:returnObject];
+                [existingUniqueKeys addObject:inputKey];
+            }
         }
         
-        [returnArray addObject:returnObject];
+        if (![returnArray containsObject:returnObject]) {
+            [returnArray addObject:returnObject];
+        }
```

Comment:
I see the test passing, but i'm not clear on why this doesn't prevent the second run through from not being added to the array or at least causing the matching object count to go to two. Can you explain a bit what's happening here?